src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -29,8 +29,12 @@ trait Iterable[+A] extends IterableOnce[A] with IterableOps[A, Iterable, Iterabl
 trait IterableOps[+A, +CC[X], +C] extends Any {
 
   protected def coll: Iterable[A]
+
   protected[this] def fromSpecificIterable(coll: Iterable[A]): C
-  protected[this] def fromIterable[E](it: Iterable[E]): CC[E]
+
+  def iterableFactory: IterableFactory[CC]
```

Comment:
Yes, I’d also like to stress that the solution proposed in this PR is very simple but does not support the use cases described in #65 and might require substantial changes to support https://github.com/scala/scala/pull/5233.